src/clientSideScene/sceneEntities.ts

```diff
@@ -1375,13 +1360,135 @@ export class SceneEntities {
       groupOfDrafts.add(groupCircle)
     }
 
-    // The target of our dragging
-    let target: Object3D | undefined = undefined
+    const insertCircle3PointKclIntoAstSnapshot = (
+      points: Vector2[]
+    ): Program => {
+      if (err(kclCircle3Point) || kclCircle3Point.program === null)
+        return kclManager.ast
+      if (kclCircle3Point.program.body[0].type !== 'ExpressionStatement')
+        return kclManager.ast
+      if (
+        kclCircle3Point.program.body[0].expression.type !== 'CallExpressionKw'
+      )
+        return kclManager.ast
+
+      // The typechecking really made this ultra unergonomic. I wrote these to
+      // help out.
+      // x should be a LabeledArg but we've got none in wasm/lang?
+      const arg = (x: any) => {
+        return (
+          (x.type === 'LabeledArg' && x.arg).type === 'ArrayExpression' &&
+          x.arg.elements
+        )
+      }
+
+      const kclCircle3PointArgs =
+        kclCircle3Point.program.body[0].expression.arguments
+
+      const arg0 = arg(kclCircle3PointArgs[0])
+      arg0[0].value = points[0].x
```

Comment:
When I use the tool I get point values with 16 decimal points; maybe it makes sense to round them to some extent?

Reply:
Yea I was thinking about it. Visually it's noisy, but technically, we shouldn't round...